Source/DafnyStandardLibraries/src/Std/Ordinal.dfy

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module Std.Ordinal {
+
+  ghost function {:axiom} Omega(): ORDINAL
+    ensures !Omega().IsNat
+    ensures Omega().IsLimit
+    ensures forall other: ORDINAL | other.IsLimit && other != 0 :: Omega() <= other
+
+  // Additional axioms about addition
+
+  lemma {:axiom} Succ(a: ORDINAL, b: ORDINAL)
+    ensures a < b <==> a + 1 <= b
+
+  ghost function {:axiom} PlusLimit(left: ORDINAL, right: ORDINAL): (result: ORDINAL)
+    decreases right, 0
+    ensures forall right' | right' < right :: Plus(left, right') < result
+    ensures forall result' {:trigger} |
+              forall right' | right' < right :: Plus(left, right') < result' :: result <= result'
```

Comment:
The fact that you're using `{:trigger}` here suggests to me that Dafny doesn't recognize that `<=` on `ORDINAL` would be acceptable as a trigger. We should consider extending Dafny to do that.

Reply:
I'd definitely love to consider adding more axioms and operations on ordinals in the future in general, but only if we think anything besides this module would benefit.